### ISSUE TITLE
Implement visit_seq too?

### ISSUE THREAD

**Issue by carols10cents** [influxdata/pbjson#83]
Currently, the `Deserializer` impls generated by pbjson [can't be used with bincode](https://github.com/influxdata/influxdb_iox/issues/6221), because [bincode uses `visit_seq`](https://github.com/bincode-org/bincode/issues/276#issuecomment-528237676) and [pbjson only implements `visit_map`](https://github.com/influxdata/pbjson/blob/5fa7d4d808983ebaf46527ddbd062d0439bad73b/pbjson-build/src/generator/message.rs#L520).

Now, this is pb**JSON**, and serde_json only uses `visit_map`, so that's definitely an argument for not generating `visit_seq` in this crate.

However, implementing it here would make the generated impls arguably more correct, make them more flexible, and be in line [with serde recommendations](https://serde.rs/deserialize-struct.html).

Are there any reasons `visit_seq` can't be implemented?

**Comment by tustvold** [comment]
> Are there any reasons visit_seq can't be implemented?

Nope, I would be very happy to review a PR that added support for this
